### PR TITLE
Fixed DP-2409

### PR DESCRIPTION
**Pull request by dmehta85**
Issue:
log filename started with a random special character.

Observation:
Issue found to be valid for ERROR, WARNING land FATAL log types but not an issue with INFO log type.

RCA:
1. Logically all things were coded right but still, we faced this issue because of the way GLOG
   accessed the arguments.
2. We pass a ‘programname’ as an argument (which declared as a local variable) from tgtd to
   HycStorCommon->folly::init->googleInitLoggigng() in new_stord REST call.
3. Here, GLOG creates an INFO type log file using that variable but not copying the value of
   an argument to its local buffer rather keeps the pointer to a variable.
4. After REST call gets completed and the value of tgtd’s local variable gets reclaimed but
   the pointer to that variable is still with GLOG and its value becomes garbage now. Till this
   time we have not logged any other type of logs apart from INFO.
5. Afterward, Whenever we try to LOG other types of logs, GLOG will try to read
   ‘programname’ from garbage variable and create the log file using that name means with
   garbage, in our case, it was a special character.

Signed-off-by: Deepen Mehta <deepen.mehta@primaryio.com>

### CODE REVIEW

File: usr/tgtd.c [hunk 1]
```diff
@@ -69,6 +69,9 @@ static pthread_mutex_t ha_rest_mutex;
 int active_rest_calls = 0;
 static pthread_mutex_t ha_active_call_cnt_mutex;
 
+char tgt_name[MAX_NAME_LEN];
+char *hyc_argv[1] = {tgt_name};
```
Comment (prasad-joshi) [inline usr/tgtd.c]: Char * hyc_argv[] = {
Tgt_name,
Null
};

File: usr/tgtd.c [hunk 2]
```diff
@@ -1009,6 +1010,7 @@ static int new_stord(const _ha_request *reqp,
 	}
 
 	pthread_mutex_lock(&ha_rest_mutex);
+	memset(tgt_name, '\0', sizeof(char)*MAX_NAME_LEN);
```
Comment (prasad-joshi) [inline usr/tgtd.c]: Memset will have no effect

File: usr/tgtd.c (hunk 1 above)
Comment (dmehta85) [reply usr/tgtd.c]: incorporated.

File: usr/tgtd.c (hunk 2 above)
Comment (dmehta85) [reply usr/tgtd.c]: incorporated.

**Comment by prasad-joshi** [comment]
@dmehta85 make sure this is merged into 1.8 as well.

**Comment by dmehta85** [comment]
Yes, I have merged the changes to both branches.

**Pull request merged**